lib/ClientHttpException.php

```diff
@@ -35,9 +33,9 @@ public function __construct(ResponseInterface $response)
     /**
      * The http status code for the error.
      */
-    public function getHttpStatus(): int
+    public function getHttpStatus(): string
     {
-        return $this->response->getStatus();
+        return (string) $this->response->getStatus();
     }
```

Comment:
See comment below - various things were "mixed" - do we consistently pass HTTP status as `string` or `int` ?